build.go

```diff
@@ -36,21 +38,33 @@ func (r *Runner) Command(command string, args ...string) error {
 	return nil
 }
 
-func (r *Runner) UpdateYum() error {
-	return r.Command("yum", "update -y")
-}
+func (r *Runner) InstallPackage(packages []string) error {
+
+	if !r.updated {
+		r.Command("yum", "update -y")
+		r.updated = true
+	}
 
-func (r *Runner) InstallPackages(packages []string) error {
 	for _, p := range packages {
-		err := r.Command("yum", "install -y "+p)
-		if err != nil {
-			return err
+		if p != "update" {
+			err := r.Command("yum", "install -y "+p)
+			if err != nil {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
 func (r *Runner) InstallGems(packages []string) error {
+
+	if !r.installed {
```

Comment:
Maybe this should be called `r.RubyInstalled`? Otherwise it's not clear without context what 'installed' we're talking about.

Reply:
this does not need to be an exported field though, right? so we can use the lowercase or camelCase convention?

Reply:
There's no _need_ to export it, but maybe it's nice to do that, just in case the user wants to be able to check it. I usually err on the side of exporting things, unless there's a good reason not to. Your call 😄